tests/sqllogictests/src/client/http_client.rs

```diff
@@ -95,6 +95,9 @@ impl HttpClient {
             let mut url = "http://127.0.0.1:8000".to_string();
             url.push_str(&next_uri);
             response = self.response(sql, &url, false).await?;
+            if response.session.is_some() {
```

Comment:
we should inject this inside `response` function.

```
fn response_inner()...

fn response() {
 let . =  
}
```

Reply:
refactored